### PR TITLE
BRMS-214: Add toggle for flag auditing

### PR DESCRIPTION
**Pull request by chris-ianson**
@adamconder Added toggle for auditing flags. Tested and when it's false the key shows up in Splunk but no values are shown.

### CODE REVIEW

File: test/uk/gov/hmrc/brm/audit/TransactionAuditorSpec.scala [hunk 1]
```diff
@@ -219,7 +219,7 @@ class TransactionAuditorSpec extends UnitSpec with MockitoSugar with OneAppPerSu
       }
 
       "return a Map() of flags where flag has reason and none" in {
-        val child1 = Record(Child(500035710, "Adam TEST", "SMITH",
+        val child1 = Record(Child(500035710: Int, "Adam TEST", "SMITH",
```
Comment (adamconder) [inline test/uk/gov/hmrc/brm/audit/TransactionAuditorSpec.scala:222]: @chris-ianson What is this doing? Int : Int?

File: app/uk/gov/hmrc/brm/audit/TransactionAuditor.scala [hunk 2]
```diff
@@ -76,7 +76,10 @@ class TransactionAuditor(connector : AuditConnector = MicroserviceGlobal.auditCo
     val auditCharactersPerNameOnRecords = recordListToMap(records, characterCount)
 
     // flags for each record
-    val auditFlags = recordListToMap(records, flags)
+    val auditFlags = BrmConfig.logFlags match {
+      case true => recordListToMap(records, flags)
+      case _ => Map().empty
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/audit/TransactionAuditor.scala:81]: @chris-ianson how does this behave with coverage?

File: test/uk/gov/hmrc/brm/audit/TransactionAuditorSpec.scala (hunk 1 above)
Comment (chris-ianson) [reply test/uk/gov/hmrc/brm/audit/TransactionAuditorSpec.scala:222]: Gets rid of the magic number warning (casting number to integer)

File: app/uk/gov/hmrc/brm/audit/TransactionAuditor.scala (hunk 2 above)
Comment (chris-ianson) [reply app/uk/gov/hmrc/brm/audit/TransactionAuditor.scala:81]: Passes

**Pull request merged**